openmc/volume.py

```diff
@@ -0,0 +1,231 @@
+from collections import Iterable, Mapping
+from numbers import Real, Integral
+from xml.etree import ElementTree as ET
+
+import numpy as np
+import pandas as pd
+
+import openmc
+import openmc.checkvalue as cv
+
+
+class VolumeCalculation(object):
+    """Stochastic volume calculation specifications and results.
+
+    Parameters
+    ----------
+    domains : Iterable of openmc.Cell, openmc.Material, or openmc.Universe
+        Domains to find volumes of
+    samples : int
+        Number of samples used to generate volume estimates
+    lower_left : Iterable of float
+        Lower-left coordinates of bounding box used to sample points. If this
+        argument is not supplied, an attempt is made to automatically determine
+        a bounding box.
+    upper_right : Iterable of float
+        Upper-right coordinates of bounding box used to sample points. If this
+        argument is not supplied, an attempt is made to automatically determine
+        a bounding box.
+
+    Attributes
+    ----------
+    ids : Iterable of int
+        IDs of domains to find volumes of
+    domain_type : {'cell', 'material', 'universe'}
+        Type of each domain
+    samples : int
+        Number of samples used to generate volume estimates
+    lower_left : Iterable of float
+        Lower-left coordinates of bounding box used to sample points
+    upper_right : Iterable of float
+        Upper-right coordinates of bounding box used to sample points
+    results : dict
```

Comment:
At the moment, the order is inconsequential. Iteration over results is used to attach volumes to cells, but in that case either the cell ID is in the results or it isn't. You'll also see in the test that I added, I specifically use `sorted(vol.results.items())` to ensure order is preserved.